resources/aws/subnet.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type Subnet struct {
+	AvailabilityZone string
+	CidrBlock        string
+	VpcID            string
+	id               string
+	AWSEntity
+}
+
+func (s *Subnet) checkIfExists() (bool, error) {
+	subnets, err := s.Clients.EC2.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: aws.String("availabilityZone"),
```

Comment:
Should we have this as a constant in the same place where `tagKeyName` is defined?